activesupport/test/json/encoding_test.rb

```diff
@@ -427,6 +427,17 @@ def test_exception_to_json
     assert_equal '"foo"', ActiveSupport::JSON.encode(exception)
   end
 
+  class DetailedException < Exception
+    def initialize(message)
+      @message = message
+    end
+  end
+
+  def test_exception_with_instance_values_to_json
+    exception = DetailedException.new("foo")
+    assert_equal %({"message":"foo"}), ActiveSupport::JSON.encode(exception)
+  end
+
```

Comment:
So it will return a string if there's no ivar, and an object if there are any?

```ruby
  class DetailedException < Exception
    def initialize(message)
      @message = message
    end
  end
ActiveSupport::JSON.encode(Exception.new("foo")) #=> '"foo"'
ActiveSupport::JSON.encode(DetailedException.new("foo")) #=> %({"message":"foo"})
```